dvc/rwlock.py

```diff
@@ -61,38 +60,25 @@ def _infos_to_str(infos):
     )
 
 
-def _check_no_writers(lock, info, path_infos):
-    for path_info in path_infos:
-        blocking_urls = lfilter(path_info.overlaps, lock["write"])
-        if not blocking_urls:
-            continue
+def _check_blockers(lock, info, *, mode, waiters):
+    for path_info in waiters:
+        blockers = [
+            info
+            for path, infos in lock[mode].items()
+            if path_info.overlaps(path)
+            if info not in (infos if type(infos) is list else [infos])
+        ]
```

Comment:
This logic is broken now. We create a list of `info` passed as param repeated 0 to several times, while in previous logic we were listing infos we've got from lock.

The right logic I guess is:
```python
blockers = [
    blocker
    for path, infos in lock[mode].items()
    if path_info.overlaps(path)
    for blocker in (infos if isinstance(infos, list) else [infos])
    if blocker != info
]
```

Reply:
Fixed in #3285.